listenbrainz/db/tests/test_user.py

```diff
@@ -18,11 +20,28 @@ def test_update_token(self):
         self.assertNotEqual(old_token, user['auth_token'])
 
     def test_update_last_login(self):
+        """ Tests db.user.update_last_login """
+
         user = db_user.get_or_create('testlastloginuser')
-        val = int(time.time())
-        db_user.update_last_login(user['musicbrainz_id'], val)
+
+        # set the last login value of the user to 0
+        with db.engine.connect() as connection:
+            connection.execute(sqlalchemy.text("""
+                UPDATE "user"
+                   SET last_login = to_timestamp(0)
+                 WHERE id = :id
+            """), {
+                'id': user['id']
+            })
+
+        user = db_user.get(user['id'])
+        self.assertEquals(int(user['last_login'].strftime('%s')), 0)
+
+        db_user.update_last_login(user['musicbrainz_id'])
         user = db_user.get_by_mb_id(user['musicbrainz_id'])
-        self.assertEqual(val, int(user['last_login'].strftime('%s')))
+
+        # after update_last_login, the val should be greater than the old value i.e 0
+        self.assertGreater(int(user['last_login'].strftime('%s')), 0)
```

Comment:
That would require changing the function to take a timestamp parameter and update to that timestamp, which is what alastair wanted changed here: https://github.com/metabrainz/listenbrainz-server/pull/189#discussion_r123195816

Initially I had written the test like you are suggesting [here](https://github.com/paramsingh/listenbrainz-server/blob/da5d86020c0fe8059c0a4a29459a56815b95391f/listenbrainz/db/tests/test_user.py#L20) but we changed it here with alastair's review: https://github.com/metabrainz/listenbrainz-server/pull/214#discussion_r125925710

Should I go back to taking a timestamp in the function and not using `NOW()`?